src/controllers/organizationController.ts

```diff
@@ -317,204 +244,339 @@ export default class OrganizationController {
-        log.error("An error occurred while signing documents. %s", error)
-      );
-      return ResponseErrorInternal("An error occurred while signing documents");
-    }
-    try {
-      await organizationInstance.update({
-        registrationStatus: OrganizationRegistrationStatusEnum.DRAFT
-      });
-    } catch (error) {
-      log.error("An error occurred updating registration status. %s", error);
-      return ResponseErrorInternal(
-        "An error occurred updating registration status"
-      );
-    }
-    try {
-      await this.emailService.send({
-        attachments: [
-          {
-            filename: "contratto.pdf",
-            path: signedContractPath
-          },
-          {
-            filename: `delega-${delegateFiscalCode.toLocaleLowerCase()}.pdf`,
-            path: signedMandatePath
-          }
-        ],
-        html:
-          localeIt.organizationController.sendDocuments.registrationEmail
-            .content,
-        subject:
-          localeIt.organizationController.sendDocuments.registrationEmail
-            .subject,
-        text:
-          localeIt.organizationController.sendDocuments.registrationEmail
-            .content,
-        to: organizationInstance.pec
-      });
-    } catch (error) {
-      log.error("An error occurred sending email. %s", error);
-      return ResponseErrorInternal("An error occurred sending email");
-    }
-    return ResponseNoContent();
+    const outputFolder = `./documents/unsigned`;
+    const createDocumentPerRequest = (request: Request) => {
+      const documentPath = `${outputFolder}/${request.id}.pdf`;
+      if (OrganizationRegistrationRequest.is(request)) {
+        return this.documentService.generateDocument(
+          request.id.toString(),
+          localeIt.organizationController.registerOrganization.contract.replace(
+            "%s",
+            `${request.organization.ipa_code} ${request.organization.fiscal_code}`
+          ),
+          documentPath
+        );
+      }
+      if (UserDelegationRequest.is(request)) {
+        return this.documentService.generateDocument(
+          request.id.toString(),
+          // TODO:
+          //  refactor this operation using an internationalization framework allowing params interpolation in strings.
+          //  @see https://www.pivotaltracker.com/story/show/169644146
+          localeIt.organizationController.registerOrganization.delegation
+            .replace(
+              "%legalRepresentative%",
+              `${request.organization.legal_representative.given_name} ${request.organization.legal_representative.family_name}`
+            )
+            .replace("%organizationName%", request.organization.name)
+            .replace(
+              "%delegate%",
+              `${request.requester.given_name} ${request.requester.family_name}`
+            ),
+          documentPath
+        );
+      }
+      return fromEither(left(new Error("Wrong data")));
```

Comment:
alternatively, you may use left imported from taskEither https://github.com/gcanti/fp-ts/blob/1.12.0/src/TaskEither.ts#L198